src/org/opensolaris/opengrok/history/GitRepository.java

```diff
@@ -197,47 +189,49 @@ Reader newLogReader(InputStream input) throws IOException {
         return new InputStreamReader(input, "UTF-8");
     }
 
-    @Override
-    public InputStream getHistoryGet(String parent, String basename, String rev) {
+    /**
+     * Try to get file contents for given revision.
+     *
+     * @param fullpath full pathname of the file
+     * @param rev revision
+     * @return contents of the file in revision rev
+     */
+    private InputStream getHistoryRev(String fullpath, String rev) {
         InputStream ret = null;
-
         File directory = new File(directoryName);
-
-        ByteArrayOutputStream output = new ByteArrayOutputStream();
-        byte[] buffer = new byte[8192];
-
         Process process = null;
+
         try {
-            String filename = (new File(parent, basename)).getCanonicalPath()
+            String filename = (new File(parent, fullpath)).getCanonicalPath()
                     .substring(directoryName.length() + 1);
             ensureCommand(CMD_PROPERTY_KEY, CMD_FALLBACK);
-            String argv[] = {RepoCommand, "show", rev + ":" + filename};
+            String argv[] = {
+                RepoCommand,
+                "show",
+                rev + ":" + filename
+            };
             process = Runtime.getRuntime().exec(argv, null, directory);
 
-            InputStream in = process.getInputStream();
-            int len;
-            boolean error = true;
-
-            while ((len = in.read(buffer)) != -1) {
-                error = false;
-                if (len > 0) {
-                    output.write(buffer, 0, len);
-                }
-            }
-            if (error) {
-                process.destroy();
-                String path = getCorrectPath(filename, rev);
-                argv[2] = rev + ":" + path;
-                process = Runtime.getRuntime().exec(argv, null, directory);
-                in = process.getInputStream();
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+            byte[] buffer = new byte[32 * 1024];
```

Comment:
I don't think so. We've had the same buffer size in MercurialRepository for a long time